internal/moduletest/file.go

```diff
@@ -17,4 +19,26 @@ type File struct {
 	Runs []*Run
 
 	Diagnostics tfdiags.Diagnostics
+
+	mu sync.Mutex
```

Comment:
You can just make the File itself implement the mutex:

```
type File struct {
  sync.Mutex

  // other fields
}
```

That way you don't need to implement the additional `Lock` function, as the you can just call lock directly on the file. Kind of optional, but this just reduces the amount of boilerplate code we need to write.

Reply:
I generally don't embed sync.Mutex, because it then exports the `Mutex` object itself, but I do agree and it makes sense here, especially since this function is entirely internal to terraform.